CsharpExtras/Map/Dictionary/Curry/CurryDictionaryRec.cs

```diff
@@ -0,0 +1,124 @@
+﻿using CsharpExtras.Extensions;
+using CsharpExtras.ValidatedType.Numeric.Integer;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace CsharpExtras.Map.Dictionary.Curry
+{
+    class CurryDictionaryRec<TKey, TVal> : CurryDictionaryBase<TKey, TVal>
```

Comment:
Maybe rename this to `CurryDictionaryRecursive`? It might make it a bit clearer.

Reply:
Done